docs/concepts/custom-ui-headless-api.mdx

```diff
@@ -0,0 +1,662 @@
+---
+id: custom-ui-headless-api
+title: Integration Basics
+sidebar_label: Integration Basics
+toc_max_heading_level: 4
+---
+
+```mdx-code-block
+import Tabs from "@theme/Tabs"
+import TabItem from "@theme/TabItem"
+import CodeBlock from "@theme/CodeBlock"
+```
+
+Ory provides headless APIs for ease of integration. This ensures that Ory provides compatibility with many software stacks across
+different programming languages.
+
+This documents' focus is to provide concrete examples on custom UI integration. Flows are the fundamental of Ory's self-service
+APIs and are used to accomplish a variety of goals, such as `login`, `registration`, `settings`, `verification` and `recovery`.
+
+The self-service `flows` are standardized so that it follows the same steps regardless of which `flow` is integrated.
+
+A general overview of the flow can be summed up in two steps:
+
+1. Get or create the `flow` data, e.g. login, registration etc.
+2. Submit the `flow` with user data (e.g. username and password)
+
+For more information on the concepts, take a look at the [Self-service flow](../kratos/self-service) documentation.
+
+## Creating a flow
+
+All browser flows can be initialized with the optional `?return_to=<url>` query parameter. This parameter is used to redirect the
+user to the specified URL after the flow is successfully completed.
+
+Some flows, such as login have additional query parameters which can be used in both browser and native applications.
+
+## Reusing an existing flow
+
+In cases where the flow already exists, we can retrieve the flow data using the flow ID. This is useful in cases where Ory has
+already initialized the flow and redirected to the UI. In such a case, the flow ID can be extracted from the URL `?flow=` query
+parameter.
+
+Here are some cases where the flow would already exist:
+
+1. Ory has initialized the flow and redirects to the UI (e.g. `/login?flow=<uuidv4>`)
+2. The request was completed by the user, however, an error occurred, and we need to retrieve the flow data to display the error
+3. The UI has 'stored' the flow ID in the URL so that page refreshes can retrieve the already existing flow data
+
+:::note
+
+The flow ID has an expiration time, so it is important to check the response status code when retrieving the flow data using an
+existing flow ID. It is up to your application to handle this error case and create a new flow.
+
+:::
+
+## Submitting a flow
```

Comment:
This section does also not explain how to submit a flow, and what happens after that. This should be added IMO.